File: lab2/source/over.cpp
#include "operations.h"
#include "factory.h"

class Over : public Operations{
public:
    void Operation(std::stack<int> &stack, std::istream& fin) override {
        Empty_check(stack);
        int previous_top = stack.top();
        stack.pop();
        Empty_check(stack);
        int tmp = stack.top();
        stack.push(previous_top);
        stack.push(tmp);
    }

};

namespace {
    Operations* createOver() {
        return new Over;
    }
    bool x = OperationsFactory::getInstance()->RegisterOperation(std::string("over"), createOver);
}

File: jaba/lab1/src/main/java/org/example/Over.java
package org.example;

import java.util.Scanner;
import java.util.Stack;

public class Over extends Operations{
    @Override
    public void Operation(Stack<Integer> stack, Scanner scanner) {
        emptyCheck(stack);
        int previous_top = stack.pop();
        emptyCheck(stack);
        int tmp = stack.peek();
        stack.push(previous_top);
        stack.push(tmp);
    }
}
